### DISCUSSION TITLE
Live Data  - parsing

### DISCUSSION THREAD

**Discussion by dkr267** [theOehrly/Fast-F1#125] [Q&A]
First and foremost, I just want to say thanks for the effort with FastF1.

My question is regards to the recording of live timing data. I have read the documentation and its quite clear that you can't do processing of the data with FastF1, until the session is finished. However, is it possible to write a script to do real time processing of the data? or is this not recommended due to the quality of the pre process data?

Thanks

**Comment by theOehrly** [comment]
The current API parser code does look-ahead and look-back to fix some problems caused by messages that are received slightly out of sync.
With this already preventing live parsing, I have not even tried to make the rest of the code capable of live running. It would require to rewrite larger parts of the code. 

I have no intention to add this feature. In my opinion, FastF1s core functionality is post-session data analysis, not live visualization.

**Reply by dkr267** [reply]
Thanks for the quick response, I was not asking for it to be included in FastF1 (as I agree it purpose is clearly defined) but more on whether parsing it (outside FastF1) provides any obviously issues?

**Reply by theOehrly** [reply]
I guess not. Apart from there being no information about the api and data format.
